pkg/controllers/webhookauthenticator/webhookauthenticator_controller.go

```diff
@@ -0,0 +1,176 @@
+	if err != nil {
+		return fmt.Errorf("failed to retrieve service openshift-oauth-apiserver/oauth-apiserver: %w", err)
+	}
+
+	kubeConfigSecret, err := c.ensureKubeConfigSecret(ctx, oauthAPIsvc, syncCtx.Recorder())
+	if err != nil {
+		return err
+	}
+
+	if authConfig.Spec.WebhookTokenAuthenticator == nil {
+		authConfig.Spec.WebhookTokenAuthenticator = &configv1.WebhookTokenAuthenticator{}
+	}
+
+	if authConfig.Spec.WebhookTokenAuthenticator.KubeConfig.Name != kubeConfigSecret.Name {
+		authConfigCopy := authConfig.DeepCopy()
+		authConfigCopy.Spec.WebhookTokenAuthenticator.KubeConfig.Name = kubeConfigSecret.Name
+		_, err := c.authentication.Update(ctx, authConfigCopy, metav1.UpdateOptions{})
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// ensureKubeConfigSecret retrieves the token from a secret for the
+// openshift-oauth-apiserver/openshift-authenticator SA. It checks that the
+// "service-ca.crt" and "token" keys are present and creates a
+// openshift-config/webhook-authentication-integrated-oauth secret with a kubeconfig
+// pointing to the oauth-apiserver's tokenvalidation endpoint
+func (c *webhookAuthenticatorController) ensureKubeConfigSecret(ctx context.Context, svc *corev1.Service, recorder events.Recorder) (*corev1.Secret, error) {
+	sa, err := c.saLister.ServiceAccounts("openshift-oauth-apiserver").Get("openshift-authenticator")
+	if err != nil {
+		return nil, err
+	}
+
+	if len(sa.Secrets) == 0 {
+		return nil, fmt.Errorf("SA openshift-authenticator does not have any tokens assigned")
+	}
+
+	var tokenSecretName string
+	for _, saSecret := range sa.Secrets {
+		if strings.Contains(saSecret.Name, "token") {
+			tokenSecretName = saSecret.Name
+		}
+	}
+
+	if len(tokenSecretName) == 0 {
+		return nil, fmt.Errorf("SA openshift-authenticator does not have any tokens assigned")
+	}
+
+	tokenSecret, err := c.secrets.Secrets("openshift-oauth-apiserver").Get(ctx, tokenSecretName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	caBundle, ok := tokenSecret.Data["service-ca.crt"]
+	if !ok {
+		return nil, fmt.Errorf("service-account token secret is missing the 'ca.crt' key")
+	}
+	token, ok := tokenSecret.Data["token"]
+	if !ok {
+		return nil, fmt.Errorf("service-account token secret is missing the 'token' key")
+	}
+
+	kubeconfigBytes, err := assets.Asset("oauth-apiserver/authenticator-kubeconfig.yaml")
+	if err != nil {
+		return nil, fmt.Errorf("failed to read kubeconfig template: %w", err)
+	}
+
+	replacer := strings.NewReplacer(
+		"${CA_DATA}", base64.StdEncoding.EncodeToString(caBundle),
+		"${APISERVER_IP}", svc.Spec.ClusterIP,
+		"${AUTHENTICATOR_TOKEN}", string(token),
+	)
+
+	kubeconfigComplete := replacer.Replace(string(kubeconfigBytes))
+
+	requiredSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "webhook-authentication-integrated-oauth",
+			Namespace: "openshift-config",
+		},
+		Data: map[string][]byte{
+			"kubeConfig": []byte(kubeconfigComplete),
```

Comment:
in the kube-apiserver, did this setting ever work before?  I doubt it.  If not, let's make the key kubeconfig

Reply:
With this PR, the authentication stack is already configured and it appears to work (if nothing else, the console-login tests pass).

Since it hasn't been documented yet, we might still change the key here, in the API and in https://github.com/openshift/cluster-kube-apiserver-operator/blob/11ffd6f568b152eb795886fd27037f7883fe77a9/pkg/operator/configobservation/auth/webhook_authenticator.go

Reply:
> Since it hasn't been documented yet, we might still change the key here, in the API and in https://github.com/openshift/cluster-kube-apiserver-operator/blob/11ffd6f568b152eb795886fd27037f7883fe77a9/pkg/operator/configobservation/auth/webhook_authenticator.go

I'd like to see it change.   This could set both for the switch.